File: Misc/DSA1/queue.py
class ListNode:
    def __init__(self, val):
        self.val = val
        self.next = None

class Queue:
    def __init__(self):
        self.head = None
        self.tail = None

    def enqueue(self, val):
        new_node = ListNode(val)
        if self.tail is None:
            self.head = new_node
            self.tail = new_node
        else:
            self.tail.next = new_node
            self.tail = new_node

    def dequeue(self):
        if self.head is None:
            raise Exception("Queue is empty")
        val = self.head.val
        self.head = self.head.next
        return val

    def print(self):
        curr = self.head
        while curr is not None:
            print(curr.val, end=" ")
            curr = curr.next
        print()

File: Misc/DSA1/queue.cs
class ListNode{
    public int val;
    public ListNode next;
    public ListNode(int val){
        this.val = val;
        this.next = null;
    }
}

class Queue{
    private ListNode head;
    private ListNode tail;

    public Queue(){
        this.head = null;
        this.tail = null;
    }

    public void Enqueue(int val){
        ListNode newNode = new ListNode(val);
        if(this.tail == null){
            this.head = newNode;
            this.tail = newNode;
        }else{
            this.tail.next = newNode;
            this.tail = newNode;
        }   
    }

    public int Dequeue(){
        if(this.head == null){
            throw new Exception("Queue is empty");
        }
        int val = this.head.val;
        this.head = this.head.next;
        return val;
    }

    public void print(){
        ListNode curr = this.head;
        while(curr != null){
            Console.Write(curr.val + " ");
            curr = curr.next;
        }
        Console.WriteLine();
    }
}